R/FilteredDataset.R

```diff
@@ -53,7 +54,7 @@ FilteredDataset <- R6::R6Class( # nolint
         }
         env <- new.env(parent = parent.env(globalenv()))
         env[[dataname]] <- private$dataset
-        filter_call <- self$get_call(sid)
+        filter_call <- private$reactive_call()
```

Comment:
It won't work for filter-counts which has non-empty `sid` as an argument. Let me explain briefly how `sid` works and why you can't cache these object in `FilteredDataset`.

For given `FilterState(s)` configuration there are two kind of calls (and data) produced in the same time. When the filter is modified possibly multiple calls of `data_filtered_fun` are invoked. This means that this can't be cached here because this function needs to be called several times with different `sid`. 

1. All FilterState are combined into one call and it is returned to the module. For this case your current solution works just perfect
<img width="625" alt="image" src="https://github.com/insightsengineering/teal.slice/assets/6959016/944cfdef-b48d-47f9-a1b2-2937aaea3122">

2. reactive counts in a single (each) FilterState are calculated on data filtered by every filter but not "this" filter.
<img width="496" alt="image" src="https://github.com/insightsengineering/teal.slice/assets/6959016/29effa59-7ab3-409d-91fc-85ad1ba6ab00">

-----

Why are counts so weird?
When you filter single column you'd like to still see the counts of the unfiltered ranges. If we don't show these counts user wouldn't know how many observations are outside of the filtered range.

| correct | incorrect |
| -------  | --------- |
| ![](https://github.com/insightsengineering/teal.slice/assets/6959016/4787fb75-f882-446a-945e-45d15e0001f9) | ![](https://github.com/insightsengineering/teal.slice/assets/6959016/80c521e3-b784-4c43-8a6e-1d6244a39969) |


Even more technical explaination [here](https://github.com/insightsengineering/teal.slice/pull/205)

<details>
<summary> example app </summary>

```r
# options(teal.log_level = "TRACE", teal.show_js_log = TRUE)
library("teal.slice")
library(teal)
library(scda)
library(SummarizedExperiment)

data <- teal_data() |> within({
  # MAE <- hermes::multi_assay_experiment
  ADSL <- synthetic_cdisc_data("latest")$adsl
  ADSL$empty <- NA
  ADSL$logical1 <- FALSE
  ADSL$logical <- sample(c(TRUE, FALSE), size = nrow(ADSL), replace = TRUE)
  ADSL$numeric <- rnorm(nrow(ADSL))
  ADSL$categorical <- sample(letters[1:10], size = nrow(ADSL), replace = TRUE)
  ADSL$date <- Sys.Date() + seq_len(nrow(ADSL))
  ADSL$datetime <- Sys.time() + seq_len(nrow(ADSL)) * 3600 * 12

  ADSL$numeric[sample(1:nrow(ADSL), size = 10, )] <- NA
  ADSL$numeric[sample(1:nrow(ADSL), size = 10, )] <- Inf
  ADSL$logical[sample(1:nrow(ADSL), size = 10, )] <- NA
  ADSL$date[sample(1:nrow(ADSL), size = 10, )] <- NA
  ADSL$datetime[sample(1:nrow(ADSL), size = 10, )] <- NA
  ADSL$categorical[sample(1:nrow(ADSL), size = 10, )] <- NA

  ADTTE <- synthetic_cdisc_data("latest")$adtte
  ADRS <- synthetic_cdisc_data("latest")$adrs
})

teal.data::datanames(data) <- c(
  #  "MAE",
  "ADSL", "ADTTE", "ADRS"
)
teal.data::join_keys(data) <- teal.data::default_cdisc_join_keys[teal.data::datanames(data)]

app <- init(
  data = data,
  modules = list(
    example_module("mod1"),
    modules(
      example_module("mod2")
    )
  ),
  filter = teal_slices(
    teal_slice("ADSL", "numeric", id = "numeric1"),
    teal_slice("ADSL", "numeric", id = "numeric2"),
    module_specific = TRUE,
    mapping = list(
      mod1 = "numeric1",
      mod2 = "numeric2"
    ),
    count_type = "all"
  )
)

runApp(app)

``` 

</details>